src/chat/Eliza.cpp

```diff
@@ -0,0 +1,421 @@
+
+// Christopher Busch
+//(c) 1993 all rights reserved.
+// eliza.cpp
+
+#include <cctype>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <search.h>
+
+#include "Database.hpp"
+#include "Eliza.hpp"
+#include "chatconstants.hpp"
+
+using namespace chat;
+
+/////YOU MAY NOT change the next 2 lines.
+const char eliza_title[] = "chat by Christopher Busch  Copyright (c)1993";
+const char eliza_version[] = "version 1.0.0";
+
+// Gets a word out of a string.
+int Eliza::get_word(const char *&input, char *outword, char &outother) {
```

Comment:
(fyi the `const char *&` is a surprising thing to callers not expecting their params to get fiddled with. Probably from the OG.

Reply:
Yes it is,  this is the hairiest stuff in the OG code. Hard to tweak it without everything breaking.